shared/utility/file/fileutil.cpp

```diff
@@ -114,7 +114,7 @@ namespace utility::file {
             closedir(dir);
         }
         else {
-            // TODO Throw an error or something
+            throw std::runtime_error("Attempted to list directory which didn't exist.\n");
```

Comment:
Don't put newlines in exceptions, it's up to the consumer of these exceptions how to format them and putting a newline in there just makes it more difficult
```suggestion
            throw std::runtime_error("Attempted to list directory which didn't exist.");
```